tests/ssl-opt.sh

```diff
@@ -5031,6 +5033,150 @@ run_test    "Record Size Limit: TLS 1.3: Server complies with record size limit
             -s "Maximum outgoing record payload length is 4095" \
             -s "10240 bytes written in 3 fragments"
 
+requires_gnutls_tls1_3
+requires_gnutls_record_size_limit
+requires_all_configs_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE MBEDTLS_SSL_CLI_C MBEDTLS_DEBUG_C
+requires_config_enabled MBEDTLS_SSL_RECORD_SIZE_LIMIT
+requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED
+run_test    "Record Size Limit: TLS 1.3: Client complies with record size limit (513), 1 fragment" \
+            "$G_NEXT_SRV --priority=NORMAL:-VERS-ALL:+VERS-TLS1.3:+CIPHER-ALL -d 4 --disable-client-cert --recordsize 512" \
+            "$P_CLI debug_level=4 force_version=tls13 request_size=256" \
+            0 \
+            -c "RecordSizeLimit: 513 Bytes" \
+            -c "ClientHello: record_size_limit(28) extension exists." \
+            -c "Sent RecordSizeLimit: 16384 Bytes" \
+            -c "EncryptedExtensions: record_size_limit(28) extension exists." \
```

Comment:
nit: rather:
```suggestion
            -c "Sent RecordSizeLimit: 16384 Bytes" \
            -c "ClientHello: record_size_limit(28) extension exists." \
            -c "RecordSizeLimit: 513 Bytes" \
            -c "EncryptedExtensions: record_size_limit(28) extension exists." \
```
On client side, we first send a record size limit extension in the ClientHello and then receive one in EncryptedExtensions.